api/extension-guides/notebook.md

```diff
@@ -0,0 +1,303 @@
+---
+# Summarize the whole topic in less than 300 characters for SEO purpose
+MetaDescription: Use the Notebook API to create rich notebook experiences within Visual Studio Code.
+---
+
+# Notebook API
+
+The notebook API allows extensions to open files as notebooks, execute notebook code cells, and render notebook outputs in a variety of rich and interactive formats. You may know of popular notebook interfaces like Jupyter Notebook or Google Colab, the notebook API allows for similar experiences inside Visual Studio Code.
+
+
+## Parts of a Notebook
+
+A notebook consists of a sequence of cells and their outputs. The cells of a notebook can be either *markdown cells* or *code cells*, and are rendered within the core of VS Code. The outputs can be of a variety of formats. Some output formats, such as plain text, JSON, images,and HTML are rendered by VS Code core. Others, such as application-specific data or interactive applets, are rendered by extensions.
+
+Cells in a notebook are read and written to disk by a `NotebookContentProvider`, which handles reading data from disk and converting it into a description of cells, as well as saving modifications to the notebook back to disk. The *code cells* of a notebook can be exectuted by a `NotebookKernel`, which provides for taking a cell and from it producing output in a variety of formats ranging from plain text to formatted documents or interactive applets. Application-specific output formats and interactive applet outputs are rendered by an `NotebookOutputRenderer`.
```

Comment:
We may want to mention it once that the content can be read-from/write-to from all kinds of storage, in addition to disk.

Reply:
Using "filesystem" as a more general form.